curriculum/challenges/english/25-front-end-development/lecture-working-with-pseudo-classes-and-pseudo-elements-in-css/672bbeb6eefd7ca9c003ea00.md

```diff
@@ -10,6 +10,260 @@ dashedName: what-are-examples-of-tree-structural-pseudo-classes
+- `:last-child` selects the last element in a parent element or the document
+- `:only-child` selects the only element in a parent element or the document
+
+Using the `:first-child` and `:last-child` pseudo-classes will select both Item 1 and Item 3 in this HTML:
+
+```html
+<ul>
+  <li>Item 1</li>
+  <li>Item 2</li>
+  <li>Item 3</li>
+</ul>
+```
+
+Here's the CSS:
+
+```css
+li:first-child {
+   background-color: orangered;
+}
+
+li:last-child {
+   background-color: lightgreen;
+}
+```
+
+If you have more unordered lists on the page, you have to be more specific with the selection: 
+
+```css
+ul li:first-child {
+   background-color: orangered;
+}
+
+ul li:last-child {
+   background-color: lightgreen;
+}
+```
+
+To show you how the `:only-child` pseudo-class works, here's an HTML example with two separate `div` elements:
+
+```html
+<div class="container">
+  <div>This is the only item in this container.</div>
+</div>
+
+<div class="container">
+  <div>This is one of two items in this container.</div>
+  <div>Here is the second item.</div>
+</div>
+```
+
+Using the `:only-child` pseudo-class ensures only the `div` element with a single child is selected:
+
+```css
+.container div:only-child {
+   border: 2px solid crimson;
+   padding: 10px;
+   background-color: lightblue;
+}
+```
+
+The `:first-of-type` and `:last-of-type` pseudo-classes select the first and last occurrence of a specific element type within its parent. They are useful for applying unique styles to the first or last instance of that element type among its siblings.
+
+In the HTML below, `:first-of-type` and `:last-of-type` applies to the first element and last element within the section element:
+
+```html
+<section>
+  <h2>Introduction</h2>
+  <p>This is the first paragraph in the first section</p>
+  <p>This is the second paragraph in the first section</p>
+</section>
+<section>
+  <h2>Details</h2>
+  <p>This is the first paragraph in the second section.</p>
+  <p>This is the second paragraph in the second section.</p>
+</section>
+```
+
+Here's the CSS:
+
+```css
+section p:first-of-type {
+   background-color: lightgreen;
+}
+
+section p:last-of-type {
+   background-color:lightblue;
+}
+```
+
+`:nth-of-type(n)` allows you to select a specific element within its parent based on its position among siblings of the same type.
+
+For instance, in the HTML below, `:nth-of-type(2)` targets the second element in the container:
+
+```html
+<div class="container">
+   <p>First paragraph</p>
+   <h2>First heading</h2>
+   <p>Second paragraph</p>
+   <p>Third paragraph</p>
+   <h2>Second heading</h2>
+</div>
+```
+
+Here's the CSS:
+
+```css
+p:nth-of-type(2) {
+    color: red;
+    font-weight: bold;
+}
+```
+
+`:only-of-type` selects an element if it’s the only one of its type within its parent. This can be useful for emphasizing single items or ensuring that they are styled differently when they’re not part of a group.
```

Comment:
```suggestion
`:only-of-type` selects an element if it's the only one of its type within its parent. This can be useful for emphasizing single items or ensuring that they are styled differently when they're not part of a group.
```